src/goto-symex/symex_main.cpp

```diff
@@ -534,3 +542,142 @@ void goto_symext::symex_step(
     UNREACHABLE;
   }
 }
+
+/// Check if an expression only contains one unique symbol (possibly repeated
+/// multiple times)
+/// \param expr: The expression to examine
+/// \return If only one unique symbol occurs in \p expr then return it;
+///   otherwise return an empty optionalt
+static optionalt<symbol_exprt>
+find_unique_pointer_typed_symbol(const exprt &expr)
```

Comment:
Can't you just use `find_symbols(const exprt &, std::set<symbol_exprt> &)` and then filter the result for pointer-typed ones?

Reply:
I didn't know about it. I originally wrote my own recursive function, but several review comments said I should do something more efficient.